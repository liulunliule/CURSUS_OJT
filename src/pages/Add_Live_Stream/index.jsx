import React, { useState } from 'react';
import "./index.scss";
import Setting from './Setting/setting';
import AddStreaming from './AddStreaming/add'

const Add_Live_Stream = () => {
  const [view, setView] = useState('addStreaming');

  const handleViewChange = (view) => {
    setView(view);
  };

  return (
    <div className='wrapper12'>
      <div className='sa4d25'>
        <div className='container-fluid'>
          <div className='row justify-content-md-center'>
            <div className='col-lg-8 col-md-10'>
              <div className='section3125'>
                <div className='justify-content-center'>
                  <div className="live-stream-page">
                    <div className="header100">
                      <button className='buttonRed100' onClick={() => handleViewChange('addStreaming')}>Add Streaming</button>
                      <button className='buttonRed100'onClick={() => handleViewChange('settings')}>Setting</button>
                    </div>
                    <div className="content100">
                      {view === 'addStreaming' && <AddStreaming />}
                      {view === 'settings' && <Setting />}
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

    
  );
}

export default Add_Live_Stream;
