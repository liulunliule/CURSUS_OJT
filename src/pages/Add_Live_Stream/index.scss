.live-stream-page {
    font-family: 'Roboto', sans-serif;
  }
  
  .header100 {
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    margin-bottom: 20px;
  }
  
  .content100 {
    padding: 20px;
    border: 1px solid #ccc;
    border-radius: 8px;
    background-color: white;
    margin-bottom: 50px;
  }
  
  .buttonRed100 {
    padding: 10px 20px;
    margin: 10px;
    margin-right: 8px;
    border: none;
    background-color: #ff4a4a;
    color: white;
    cursor: pointer;
    border-radius: 4px;
    font-size: 14px;
    // margin-top: 0;
    font-weight: 500 !important;
    font-family: 'Roboto', sans-serif;
    

    &:hover {
        background-color: #e04343;
    }
  }
  
  .inputRead {
    width: 80%;
    padding: 10px;
    margin: 10px 0;
    border: 1px solid #ccc;
    border-radius: 4px;
  }
  
  .randomLine {
    display: flex;
    align-items: center;
    margin: 10px 0;
    font-weight: 400;
  }
  
  .inputToggle {
    margin-right: 10px;
  }

  .live-stream-title {
        font-size: 18px;
        font-weight: 500;
        font-family: 'Roboto', sans-serif;
        text-align: center;
        color: #333;
        line-height: 24px;
        margin-bottom: 0;
    
  }

.whatever {
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    color: #686f7a;
    text-align: center;
    margin-top: 5px;
    line-height: 24px;
    margin-bottom: 20px;
}

.learn-more {
    color: #333;
    font-weight: 500;
    font-size: 14px;
    font-family: 'Roboto', sans-serif;
    margin-left: 4px;
}

.sth-imp {
    font-size: 18px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    margin-top: 30px;
    color: #333;
    line-height: 24px;
}

.group-form {
    margin-top: 27px;
    position: relative;
}

.group-form label {
    font-size: 14px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    margin-bottom: 10px;
    color: #333;
    line-height: 24px;
}

._dlor1 {
    border: 1px solid #efefef;
    background: #f7f7f7;
    height: 40px;
    width: 100%;
    padding: 0 100px 0 20px;
    border-radius: 3px;
    font-family: 'Roboto', sans-serif;
    font-weight: 400;
    font-size: 13px;
}

._6tf7s {
    position: absolute;
    right: 5px;
    bottom: 5px;
    height: 30px;
    padding: 0 15px;
    background: #ed2a26;
    color: #fff;
    border: 0;
    border-radius: 3px;
    font-size: 13px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
}