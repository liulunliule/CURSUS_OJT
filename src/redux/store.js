import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
