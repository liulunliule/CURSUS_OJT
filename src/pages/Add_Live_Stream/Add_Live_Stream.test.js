import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import Add_Live_Stream from '.';

test('test verification page', () => {
  const tree = renderer.create(
    <Provider store={store}>
        <MemoryRouter>
            <Add_Live_Stream />
        </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});