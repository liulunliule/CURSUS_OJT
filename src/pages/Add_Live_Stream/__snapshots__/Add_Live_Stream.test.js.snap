// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test verification page 1`] = `
<div
  className="wrapper12"
>
  <div
    className="sa4d25"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row justify-content-md-center"
      >
        <div
          className="col-lg-8 col-md-10"
        >
          <div
            className="section3125"
          >
            <div
              className="justify-content-center"
            >
              <div
                className="live-stream-page"
              >
                <div
                  className="header100"
                >
                  <button
                    className="buttonRed100"
                    onClick={[Function]}
                  >
                    Add Streaming
                  </button>
                  <button
                    className="buttonRed100"
                    onClick={[Function]}
                  >
                    Setting
                  </button>
                </div>
                <div
                  className="content100"
                >
                  <div
                    className="add-streaming"
                  >
                    <h2
                      className="live-stream-title"
                    >
                      Connect your live stream to the Live API
                    </h2>
                    <div
                      className="whatever"
                    >
                      Use live-streaming software or a hardware encoder.
                      <a
                        className="learn-more"
                        href="#"
                      >
                        Learn More
                      </a>
                    </div>
                    <div
                      className="sth-imp"
                    >
                      Preview your broadcast with a stream key or paired encoder.
                    </div>
                    <div
                      className="group-form"
                    >
                      <label>
                        Server URL*
                      </label>
                      <input
                        className="_dlor1"
                        readOnly={true}
                        type="text"
                        value="rtmps://live-api-s.edututs+.com:443/rtmp/"
                      />
                      <button
                        className="_6tf7s"
                      >
                        Copy
                      </button>
                    </div>
                    <div
                      className="group-form mb-20"
                    >
                      <label>
                        Persistent stream key*
                      </label>
                      <input
                        className="_dlor1"
                        readOnly={true}
                        type="text"
                        value="592030151361629?s_bl=1&s_ps=1&s_sw=065&s_vt=api&s_a=AbzB7xYk7XdnKpBf"
                      />
                      <button
                        className="_6tf7s"
                      >
                        Copy
                      </button>
                    </div>
                    <button
                      className="buttonRed100"
                    >
                      Go Live
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
